core/dbt/cli/options.py

```diff
@@ -12,6 +16,19 @@ def __init__(self, *args, **kwargs):
         self._previous_parser_process = None
         self._eat_all_parser = None
 
+        # validate that multiple=True
+        multiple = kwargs.pop("multiple", None)
+        msg = f"MultiOption named `{self.name}` must have multiple=True (rather than {multiple})"
+        assert multiple, msg
+
+        # validate that type=tuple or type=ChoiceTuple
+        option_type = kwargs.pop("type", None)
+        msg = f"MultiOption named `{self.name}` must be tuple or ChoiceTuple (rather than {option_type})"
+        if inspect.isclass(option_type):
+            assert issubclass(option_type, tuple), msg
+        else:
+            assert isinstance(option_type, ChoiceTuple), msg
```

Comment:
Very open to feedback here -- I just stuffed quick-n-dirty validation in the easiest place possible.